### PR TITLE
Fix general issues with breaking things

### PR DESCRIPTION
**Pull request by bruberu**
This PR fixes two minor issues I've found with breaking blocks. One of these is that the server does not correctly deal with a broken block when a sound is played, which has been fixed by adding parameters to onBreakingUse to allow the server to position a sound to where the tool is applied and then play it to everyone. The other is to fix an issue with null MTEs when they are broken at a low TPS, as they would cause persistent visual errors on the client side when broken, which has simply been fixed by turning them into air on the client on update cycles. It appears that they can return to normal MTEs if the server syncs its data properly. However, they seem to still potentially appear for a very short time, although for much shorter durations than previously.

### CODE REVIEW

File: src/main/java/gregtech/api/metatileentity/MetaTileEntityHolder.java [hunk 1]
```diff
@@ -128,7 +128,10 @@ public boolean hasCapability(@Nonnull Capability<?> capability, @Nullable EnumFa
     public void update() {
         if (metaTileEntity != null) {
             metaTileEntity.update();
+        } else if (world.isRemote) { // Remove null TEs on the client
+            this.getWorld().setBlockToAir(this.getPos());
         }
+
```
Comment (Yefancy) [inline src/main/java/gregtech/api/metatileentity/MetaTileEntityHolder.java:134]: This code is simple, but it makes a big impact. I wonder how you test it? 
- How to ensure data synch between the server and the client? 
- How you test it in low tps that you mentioned?
- `It appears that they can return to normal MTEs if the server syncs its data properly.` How is this confirmed?

I think the core problem is to solve why MTE is synced to empty, but `BlockMachine` not.

**Comment by Yefancy** [comment]
A quick look at the MetaTileEntityHolder code shows that I can't see when a null metaTileEntity is assigned. 
probably a fake MTEHolder was accessed while rendering, is there any way to reproduce this issue? I think I can try to fix it in the renderer.

**Comment by Yefancy** [comment]
btw, the cheapest way is to have the `BlockMachine` return a non-full CollisionBox, which at least ensures that the surrounding blocks are rendered as normal. also do some check with the onDrawBlockHighlight event.

File: src/main/java/gregtech/api/metatileentity/MetaTileEntityHolder.java (hunk 1 above)
Comment (bruberu) [reply src/main/java/gregtech/api/metatileentity/MetaTileEntityHolder.java:134]: 1. I have noticed several times that if this accidentally hits, the server seems to be able to recover it; besides, I would think that TE updates on the server go to the client, which could recover the block itself. I've also noticed an occasional flicker sometimes in low TPS while placing blocks, but it always properly recovers.
 2. I put a Thread.sleep of 1 in MetaTileEntity's update, which quite easily makes a nice configurable lag.
 3. Again, it also looks like placing is made a little weird in low TPS environments, and I can see the MTE flicker for a tick, and then return, as if the client code deleted the block, but the server returned it.

**Comment by bruberu** [comment]
> btw, the cheapest way is to have the `BlockMachine` return a non-full CollisionBox, which at least ensures that the surrounding blocks are rendered as normal. also do some check with the onDrawBlockHighlight event.

This wouldn't do much overall to solve the issue, and besides, null MTEs almost certainly cause other problems than just its rendering on the client side. It's likely better to just remove these side effects immediately.

File: src/main/java/gregtech/api/metatileentity/MetaTileEntityHolder.java (hunk 1 above)
Comment (Yefancy) [reply src/main/java/gregtech/api/metatileentity/MetaTileEntityHolder.java:134]: As you said, I simulated a low TPS environment (4). But didnt recovering, with a completely invisible block blocking me.  
worse2
.
The good news is that I seem to have figured out how to reproduce invisible blocks inspired by you and have some ideas.

**Review by Yefancy — APPROVED** [review]

File: src/main/java/gregtech/api/net/NetworkHandler.java [hunk 2]
```diff
@@ -223,29 +223,9 @@ public static void init() {
...
-                    if(te instanceof MetaTileEntityHolder && ((MetaTileEntityHolder) te).getMetaTileEntity() instanceof MetaTileEntityMonitorScreen) {
-                        MonitorPluginBaseBehavior pluginBaseBehavior = ((MetaTileEntityMonitorScreen) ((MetaTileEntityHolder) te).getMetaTileEntity()).plugin;
-                        if (pluginBaseBehavior != null) {
-                            return new CPacketPluginSynced(pluginBaseBehavior, buf);
-                        }
-                    }
-                    return new CPacketPluginSynced(null, buf);
-                }
-        ));
+        CPacketPluginSynced.registerPacket(7);
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/api/net/NetworkHandler.java:226]: Maybe move these packet numbers to `GregtechDataCodes`?

File: src/main/java/gregtech/api/block/machines/BlockMachine.java [hunk 3]
```diff
@@ -177,6 +175,11 @@ public float getExplosionResistance(@Nonnull World world, @Nonnull BlockPos pos,
         return collisionList;
     }
 
+    @Override
+    public boolean doesSideBlockRendering(@Nonnull IBlockState state, @Nonnull IBlockAccess world, @Nonnull BlockPos pos, @Nonnull EnumFacing face) {
+        return state.isOpaqueCube() && getMetaTileEntity(world, pos) != null;
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/api/block/machines/BlockMachine.java:180]: Should this have an additional check for if the face to be rendered is obscured by another block?

Comment (Yefancy) [reply src/main/java/gregtech/api/block/machines/BlockMachine.java:180]: this is actually a method called by another block to check side rendering.

Comment (ALongStringOfNumbers) [reply src/main/java/gregtech/api/block/machines/BlockMachine.java:180]: ah, I see

File: src/main/java/gregtech/api/net/NetworkHandler.java (hunk 2 above)
Comment (Yefancy) [reply src/main/java/gregtech/api/net/NetworkHandler.java:226]: blueberry said he will rework a general refactoring of NetworkHandler.

**Review by ALongStringOfNumbers — APPROVED** [review]
Since both of my concerns were addressed, I think it will be fine to merge this and see if any problems are reported, since it is hard to test

**Review by TechLord22 — APPROVED** [review]

**Pull request merged**